collect_app/src/main/java/org/odk/collect/android/application/Collect.java

```diff
@@ -148,41 +131,14 @@ public void onCreate() {
         singleton = this;
 
         setupDagger();
-
-        NotificationUtils.createNotificationChannel(singleton);
+        applicationInitializer.initializePreferences();
+        applicationInitializer.initializeFrameworks();
+        applicationInitializer.initializeLocale();
 
         registerReceiver(new SmsSentBroadcastReceiver(), new IntentFilter(SMS_SEND_ACTION));
         registerReceiver(new SmsNotificationReceiver(), new IntentFilter(SMS_NOTIFICATION_ACTION));
 
-        try {
-            JobManager
-                    .create(this)
-                    .addJobCreator(collectJobCreator);
-        } catch (JobManagerCreateException e) {
-            Timber.e(e);
-        }
-
-        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
-        FormMetadataMigrator.migrate(prefs);
-        PrefMigrator.migrateSharedPrefs();
-        AutoSendPreferenceMigrator.migrate();
-
-        reloadSharedPreferences();
-
-        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
-        JodaTimeAndroid.init(this);
-
-        defaultSysLanguage = Locale.getDefault().getLanguage();
-        new LocaleHelper().updateLocale(this);
-
         initializeJavaRosa();
```

Comment:
Oh I'd meant to come back to these! JavaRosa is a little complex as `initializeJavaRosa` is called in lots of places but I think we should be able to include OSMDroid at least. I'll look at this again.

Reply:
OK moved the map stuff in (the OSM problem we were seeing in Robolectric before has apparently gone now which is nice). I decided to keep the receivers and the JavaRosa in `onCreate`. For the former it's because they are directly linked to the application object itself. For the latter it's because that method is called in many places and I think that's a rework for another time 😁